packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -151,6 +187,21 @@ export default function __experimentalUseColors(
 		[ setAttributes, colorConfigs.length ]
 	);
 
+	const detectedBackgroundColorRef = useRef();
+	const BackgroundColorDetector = useMemo(
```

Comment:
We are calling withFallbackStyles, even if the block does not use the detection functionality or the contrast functionality at all. Given that the reference is undefined, the withFallbackStyles will not do much. But, I still wonder, could we make these calls only if the background detection functionality is used?
We would an auxiliary component given than hooks always need to be called in the same order.

Reply:
Not in any straightforward way, because we want to return a component, not an element so that it can take props easily.

I made it so that `withFallbackStyles` is only called when needed so that we avoid the potentially expensive part which is creating the component.